js/libs/core.js

```diff
@@ -67,6 +67,76 @@ Fliplet.FormBuilder = (function() {
         component.methods = {};
       }
 
+      component.methods.setSourceValue = function(data) {
+        var result;
+        var $vm = this;
+
+        switch (data.source) {
+          case 'profile':
+            if (!data.key) {
+              throw new Error('A key is required to fetch data from the user\'s profile');
+            }
+
+            result = Fliplet.User.getCachedSession();
+            break;
+          case 'query':
+            if (!data.key) {
+              throw new Error('A key is required to fetch data from the navigation query parameters');
+            }
+
+            result = Fliplet.Navigate.query[data.key];
+            break;
+          case 'appStorage':
+            if (!data.key) {
+              throw new Error('A key is required to fetch data from the storage');
+            }
+
+            result = Fliplet.App.Storage.get(data.key);
+            break;
+          default:
+            result = this.value;
+        }
+
+        if (!(result instanceof Promise)) {
+          result = Promise.resolve(result);
+        }
+
+        return result.then(function (value) {
+          if (typeof value === 'undefined') {
+            value = '';
+          }
+
+          if (typeof value === 'object') {
+            if (value && value.entries) {
```

Comment:
This bit of the logic really only applies to when `result` comes from `Fliplet.User.getCachedSession()`, so I'd just add it as a `.then()` after the `Fliplet.User.getCachedSession()`.